File: prog/1stPartyLibs/quirrel/quirrel/doc/source/performance/tests/lua/fib_loop.lua
local function fibI(n)
    local last = 0
    local cur = 1
    n = n - 1
    while (n > 0)
    do
        n = n - 1
        local tmp = cur
        cur = last + cur
        last = tmp
    end
    return cur
end
loadfile("profile.lua")()

io.write(string.format("\"fibonacci loop\", %.8f, 20\n", profile_it(20, function () fibI(6511134) end)))

File: prog/1stPartyLibs/quirrel/quirrel/bench/js/fib_loop.js
// fibonacii test

function fibI(n) {
	var last = 0
	var cur = 1
	n = n - 1
	while ( n>0 ) {
		n = n - 1
		var tmp = cur
		cur = last + cur
		last = tmp
	}
	return cur
}

// infrastructure

function timeStamp() {
	return Date.now();
}


function profile(tname,cnt,testFn) {
	var t = 100500;
	var count = cnt
	while ( count>0 ) {
		var t0 = timeStamp();
		testFn();
		var t1 = timeStamp();
		var dt = t1 - t0;
		t = Math.min(dt, t);
		count --;
	}
	t /= 1000.0;
	var msg = '"'+tname+'", '+t+', '+cnt;
	print(msg)
}

function performance_tests() {
	profile("fibonacci loop",20,function(){
		fibI(6511134);
	});
	timeStamp();
}

performance_tests();
